### PR TITLE
[pt1] Supporting work for getting AES on to kubebuilder

### PR DESCRIPTION
**Pull request by LukeShu**
## Description
This is making a few tweaks to the Go CRD types, to make them more consumable by apro.git.  The most notable thing here is that I've introduced a `MillisecondDuration` type (`+kubebuilder:validation:Type="integer"`) so that all of the `{whatever}_ms` fields can be typed a little better than `int`.

## Related Issues
 1. emissary: https://github.com/emissary-ingress/emissary/pull/3939 <-- this PR
 2. apro-stubs: https://github.com/datawire/apro-stubs/pull/5
 3. aes-ratelimit-1.3: https://github.com/datawire/aes-ratelimit/pull/18
 4. aes-ratelimit-1.4: https://github.com/datawire/aes-ratelimit/pull/19
 5. edge-stack: https://github.com/datawire/edge-stack/pull/30
 6. apro-1.14: https://github.com/datawire/apro/pull/2851
 7. apro-2.1: https://github.com/datawire/apro/pull/2848

## Testing
A few sentences describing what testing you've done, e.g., manual tests, automated tests, deployed in production, etc.

## Checklist

 - [x] I made sure to update `CHANGELOG.md`. - no applicable changes
   
   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations
 
 - [x] This is unlikely to impact how Ambassador performs at scale.
 
   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability
 
 - [x] My change is adequately tested. - CI should cover it
 
   Remember when considering testing:
    + LEGACY MODE TESTS DO NOT RUN FOR EVERY PR. If your change is affected by legacy mode, you need
      to run legacy-mode tests manually (set `AMBASSADOR_LEGACY_MODE=true` and run the tests).
      (This will be fixed soon.)
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points
 
 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently. - no tricks

### CODE REVIEW

File: pkg/api/getambassador.io/v3alpha1/authservice_types.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type AuthServiceIncludeBody struct {
 
 // Why isn't this just an int??
```
Comment (kflynn) [inline pkg/api/getambassador.io/v3alpha1/authservice_types.go:35]: Probably should drop this comment sometime. 😂  (Also, thank you. 🙂 )

**Review by kflynn — DISMISSED** [review]
Hit it! (There's a comment that can be deleted, but, no need to do it now and screw up all the other PRs.)

File: pkg/api/getambassador.io/v3alpha1/authservice_types.go (hunk 1 above)
Comment (LukeShu) [reply pkg/api/getambassador.io/v3alpha1/authservice_types.go:35]: No, that comment is still valid. But I'll reword it to be clearer.

**Review by kflynn — APPROVED** [review]
This is great. Hit it! 🙂

**Review by aidanhahn — APPROVED** [review]
